js/views/stock.js

```diff
@@ -1 +1,41 @@
-// define your StockView here
+var StockView = function(page){
+	this.view = document.querySelector(page);
+	this.initialize();
+	this.updateShares();
+
+}
+
+StockView.prototype.initialize = function(){
+	document.getElementById("find-symbol").addEventListener("submit", this.submitEventHandler.bind(this));
+}
+
+StockView.prototype.updateShares = function(){
+	document.getElementById("find-shares").addEventListener("submit", this.submitEventHandlerShares.bind(this));
+}
+
+StockView.prototype.submitEventHandlerShares = function(event){
+	event.preventDefault();
+	var sharesValue = parseInt(document.getElementById("new-number-shares-text").value);
+	console.log("updating shares", sharesValue);
+	this.sharesMath(sharesValue);
+}
+
+StockView.prototype.sharesMath = function(sharesValue){
+	this.model.updateShares(sharesValue);
```

Comment:
This is pretty complicated! I had to juggle `Stock.updateShares` and `StockView.updateShares` and `StockView.submitEventHandlerShares` and `StockView.sharesMath` to figure out what was going on.